dds/DCPS/RecorderImpl.cpp

```diff
@@ -228,13 +228,24 @@ void RecorderImpl::data_received(const ReceivedDataSample& sample)
 
   // we only support SAMPLE_DATA messages
   if (sample.header_.message_id_ == SAMPLE_DATA && listener_.in()) {
+    Encoding::Kind kind = Encoding::KIND_UNALIGNED_CDR;
+    if (sample.header_.cdr_encapsulation_) {
+      Encoding enc;
+      Serializer ser(sample.sample_.get(), enc);
+      EncapsulationHeader encap;
+      if (ser >> encap) {
+        encap.to_encoding(enc, FINAL, false);
+        kind = enc.kind();
+      }
+    }
```

Comment:
Could we have a setting in the `Recorder` to skip this? That would be a simple way leave it open to be manually parsed by the user.